mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
> The problem with hashing is they're unpredictable. This way, if a stimulus is called "S196" or "D196", it will be 196 - this is, I think, the least surprising solution. You don't have to look up what hash 8sj@us is, you just know "S196" == 196.

okay, how about we allow `event_id`. If it's `None`, use hashes. But otherwise, the user can write two lines of code to get the least surprising solution they want by passing it as `event_id`. 

I'm worried that this will break. If you can make it work for all possible strings (add tests with different strings as event_ids) + keep the code simple, I'm fine with your solution too.